packages/nodejs/src/transmitter.ts

```diff
@@ -4,79 +4,171 @@ import http from "http"
+  onError: (error: Error) => void
+}
 
 export class Transmitter {
   #config: Configuration
   #url: string
-  #data: string
+  #body: string
 
-  constructor(url: string, data = "") {
+  constructor(url: string, body = "") {
     this.#config = new Configuration({})
     this.#url = url
-    this.#data = data
+    this.#body = body
   }
 
-  public async transmit() {
-    return new Promise<HashMap<any>>((resolve, reject) => {
-      const config_data = this.#config.data
-      const params = new URLSearchParams({
-        api_key: config_data["pushApiKey"] || "",
-        name: config_data["name"] || "",
-        environment: config_data["environment"] || "",
-        hostname: config_data["hostname"] || ""
-      })
+  public download() {
+    return new Promise<http.IncomingMessage>((resolve, reject) => {
+      this.request({
+        method: "GET",
+
+        callback(stream) {
+          const statusCode = stream.statusCode!
+
+          if (statusCode >= 400) {
+            reject({ kind: "statusCode", statusCode })
+          } else {
+            resolve(stream)
+          }
+        },
 
-      const url = new URL(`${this.#url}?${params.toString()}`)
-      const opts = this.requestOptions(url)
-      const requestModule = url.protocol == "http:" ? http : https
-
-      const request = requestModule.request(opts, (stream: any) => {
-        const responseStatus = stream.statusCode
-        stream.setEncoding("utf8")
-        let responseBody = ""
-
-        stream
-          .on("data", (chunk: any) => {
-            responseBody += chunk
-          })
-          .on("end", () => {
-            let parsedResponse
-            try {
-              parsedResponse = JSON.parse(responseBody)
-            } catch {
-              parsedResponse = {}
-            }
-
-            resolve({ status: responseStatus, body: parsedResponse })
-          })
+        onError(error) {
+          reject({ kind: "requestError", error })
+        }
       })
+    })
+  }
+
+  public async transmit() {
+    return new Promise<{ status: number; body: any }>((resolve, reject) => {
+      this.request({
+        method: "POST",
+        params: this.configParams(),
+
+        callback(stream) {
+          const responseStatus = stream.statusCode!
+          stream.setEncoding("utf8")
+          let responseBody = ""
+
+          stream
+            .on("data", (chunk: any) => {
+              responseBody += chunk
+            })
+            .on("end", () => {
+              let parsedResponse
+              try {
+                parsedResponse = JSON.parse(responseBody)
+              } catch {
+                parsedResponse = {}
+              }
 
-      request.on("error", error => {
-        reject({ error: error })
+              resolve({ status: responseStatus, body: parsedResponse })
+            })
+        },
+
+        onError(error) {
+          reject({ error: error })
+        }
       })
+    })
+  }
+
+  public request({
```

Comment:
```suggestion
  private request({
```
If possible I'd like to make this private. We probably shouldn't call it directly.

Reply:
It's not possible to write tests against it if it's private. I wouldn't want to duplicate the tests for the CA behavior between `.transmit` and `.downloadStream`.